invenio_communities/assets/semantic-ui/js/invenio_communities/subcommunity/search.js

```diff
@@ -6,47 +6,49 @@
  * under the terms of the MIT License; see LICENSE file for more details.
  */
 
-import { createSearchAppInit } from "@js/invenio_search_ui";
+import React from "react";
+import ReactDOM from "react-dom";
 import {
   ContribBucketAggregationElement,
   ContribBucketAggregationValuesElement,
   ContribSearchAppFacets,
 } from "@js/invenio_search_ui/components";
-import { overrideStore, parametrize } from "react-overridable";
+import { parametrize } from "react-overridable";
 import {
-  CommunitiesResults,
   CommunitiesSearchBarElement,
-  CommunitiesSearchLayout,
   CommunityItem,
   ResultsGridItemTemplate,
+  CommunitySelectionSearch,
 } from "../community";
 
-const appName = "InvenioSubCommunities.Search";
+import { Container } from "semantic-ui-react";
+import { SubcommunityResults } from "./SubcommunityResults";
 
 const ContribSearchAppFacetsWithConfig = parametrize(ContribSearchAppFacets, {
   help: false,
 });
 
-const CommunitiesSearchLayoutConfig = parametrize(CommunitiesSearchLayout, {
-  appName: appName,
-});
-
-export const defaultComponents = {
-  [`${appName}.BucketAggregation.element`]: ContribBucketAggregationElement,
-  [`${appName}.BucketAggregationValues.element`]: ContribBucketAggregationValuesElement,
-  [`${appName}.SearchApp.facets`]: ContribSearchAppFacetsWithConfig,
-  [`${appName}.ResultsList.item`]: CommunityItem,
-  [`${appName}.ResultsGrid.item`]: ResultsGridItemTemplate,
-  [`${appName}.SearchApp.layout`]: CommunitiesSearchLayoutConfig,
-  [`${appName}.SearchBar.element`]: CommunitiesSearchBarElement,
-  [`${appName}.SearchApp.results`]: CommunitiesResults,
+export const overriddenComponents = {
+  [`BucketAggregation.element`]: ContribBucketAggregationElement,
+  [`BucketAggregationValues.element`]: ContribBucketAggregationValuesElement,
+  [`SearchApp.facets`]: ContribSearchAppFacetsWithConfig,
+  [`ResultsList.item`]: CommunityItem,
+  [`ResultsGrid.item`]: ResultsGridItemTemplate,
+  [`SearchBar.element`]: CommunitiesSearchBarElement,
+  [`SearchApp.results`]: SubcommunityResults,
 };
 
-const overriddenComponents = overrideStore.getAll();
-// Auto-initialize search app
-createSearchAppInit(
-  { ...defaultComponents, ...overriddenComponents },
-  true,
-  "invenio-search-config",
-  true
+const domContainer = document.getElementById("communities-search");
+const config = JSON.parse(domContainer.dataset?.invenioSearchConfig);
+const userAnonymous = JSON.parse(domContainer.dataset?.userAnonymous);
+// eslint-disable-next-line react/no-deprecated
+ReactDOM.render(
+  <Container>
+    <CommunitySelectionSearch
```

Comment:
I had to create a component due to the changes of `config`, based on the selection of "All" or "My communities". I could not connect the config with the component returned by `createSearchAppInit`, so I created a component that instantiates the search application.